### PR TITLE
chore(hybrid-cloud): Break AuditLogEntry and Activity foreign keys

### PR DESCRIPTION
**Pull request by corps**
Breaks AuditLogEntry and Activity foreign keys that span silos.  Migration is pretty safe -- just a constant time constraint removal.

### CODE REVIEW

File: src/sentry/models/activity.py [hunk 1]
```diff
@@ -89,7 +93,8 @@ class Activity(Model):
     type = BoundedPositiveIntegerField(choices=CHOICES)
     ident = models.CharField(max_length=64, null=True)
     # if the user is not set, it's assumed to be the system
-    user = FlexibleForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete=models.SET_NULL)
+    # user = FlexibleForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete=models.SET_NULL)
```
Comment (dashed) [inline src/sentry/models/activity.py:96]:
```suggestion
```

File: src/sentry/admin.py [hunk 2]
```diff
@@ -325,12 +325,12 @@ def response_add(self, request, obj, post_url_continue=None):
...
-    list_filter = ("event", "datetime")
-    search_fields = ("actor__email", "organization__name", "organization__slug")
-    raw_id_fields = ("organization", "actor", "target_user")
+    list_display = ("event", "organization_id", "actor", "datetime")
+    list_filter = ("event", "datetime", "organization_id")
+    search_fields = ("actor__email",)
+    raw_id_fields = ("actor", "target_user")
     readonly_fields = (
-        "organization",
+        "organization_id",
```
Comment (markstory) [inline src/sentry/admin.py:333]: There are other models in this module that also have `user` and `organization` in the `readonly_fields` and `raw_id_fields` lists. Do those also need to be updated?

File: src/sentry/utils/audit.py [hunk 3]
```diff
@@ -42,10 +42,19 @@ def create_audit_entry_from_user(
     ip_address: str | None = None,
     transaction_id: int | str | None = None,
     logger: Logger | None = None,
+    organization: Organization | None = None,
+    organization_id: int | None = None,
     **kwargs: Any,
 ) -> AuditLogEntry:
+    if organization:
+        organization_id = organization.id
```
Comment (markstory) [inline src/sentry/utils/audit.py:50]: Do we want to guard against the scenario where both `organization` and `organization_id` are provided at the same time?

File: src/sentry/utils/audit.py [hunk 4]
```diff
@@ -146,13 +157,20 @@ def _complete_delete_log(delete_log: DeletedEntry, entry: AuditLogEntry) -> None
...
+    organization_id: int | None = None,
+    **kwargs: Any,
 ) -> AuditLogEntry:
     """
     Creates an audit log entry for events that are triggered by Sentry's
     systems and do not have an associated Sentry user as the "actor".
     """
-    entry = AuditLogEntry(actor_label="Sentry", **kwargs)
+    if organization:
+        organization_id = organization.id
```
Comment (markstory) [inline src/sentry/utils/audit.py:171]: Same here with guarding against both `organization` and `organization_id` being provided together.

File: src/sentry/admin.py (hunk 2 above)
Comment (corps) [reply src/sentry/admin.py:333]: For sure, when their foreign key gets broken, but I wanted to ensure the changesets followed along the keys I break incrementally than make the full change all at once.

File: src/sentry/utils/audit.py (hunk 3 above)
Comment (corps) [reply src/sentry/utils/audit.py]: Added in latest commit

**Review by markstory — APPROVED** [review]

**Pull request merged**